plugins/hdfsviewer/src/azkaban/viewer/hdfs/ORCFileViewer.java

```diff
@@ -0,0 +1,153 @@
+package azkaban.viewer.hdfs;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.EnumSet;
+import java.util.Set;
+
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.hive.ql.io.orc.OrcFile;
+import org.apache.hadoop.hive.ql.io.orc.Reader;
+import org.apache.hadoop.hive.ql.io.orc.RecordReader;
+import org.apache.log4j.Logger;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import azkaban.viewer.hdfs.utils.SerDeUtilsWrapper;
+
+/**
+ * This class implements a viewer for ORC files
+ *
+ * @author gaggarwa
+ */
+public class ORCFileViewer extends HdfsFileViewer {
+    private static Logger logger = Logger.getLogger(ORCFileViewer.class);
+    // Will spend 5 seconds trying to pull data and then stop.
+    private final static long STOP_TIME = 5000l;
+    private final static int JSON_INDENT = 2;
+
+    private static final String VIEWER_NAME = "ORC";
+
+    @Override
+    public String getName() {
+        return VIEWER_NAME;
+    }
+
+    /**
+     * Get ORCFileViewer functionalities. Currently schema and read are
+     * supported. {@inheritDoc}
+     *
+     * @see azkaban.viewer.hdfs.HdfsFileViewer#getCapabilities(org.apache.hadoop.fs.FileSystem,
+     *      org.apache.hadoop.fs.Path)
+     */
+    @Override
+    public Set<Capability> getCapabilities(FileSystem fs, Path path) {
+        if (logger.isDebugEnabled()) {
+            logger.debug("Parquet file path: " + path.toUri().getPath());
+        }
+
+        Reader orcReader = null;
+        RecordReader recordReader = null;
+        try {
+            // no need to close orcreader
+            orcReader = OrcFile.createReader(fs, path);
+            recordReader = orcReader.rows(null);
+        } catch (Exception e) {
+            if (logger.isDebugEnabled()) {
```

Comment:
@hluu Theoretically there should not be a scenario but we rely on how good hive orc reader is.  

Basically we need a mechanism to assign a Fileviewer to a file and Azkaban uses this exception mechanism. I would think of this exception as a particular specific FileViewer is not capable of viewing the given file. Otherwise we need to change this  behaviour for all FileViewer on the basis of other param e.g. extension